### PR TITLE
Allow VS Code to recognise files with "sourcekit-lsp" scheme to provide Semantic Functionality through SourceKitLSP

### PR DESCRIPTION
**Pull request by lokesh-tr**
This PR makes VS Code to recognise files with scheme "sourcekit-lsp", thereby allowing them to be handled by SourceKitLSP and provide Semantic Functionality. Specifically, This PR enables Semantic Functionality in Macro Expansion Reference Document's Peeked Editors.

Refer to the sourcekit-lsp PR below for more information on what semantic functionality is supported as of now.

Accompanying PR in sourcekit-lsp repository: https://github.com/swiftlang/sourcekit-lsp/pull/1610

------
[Expansion of Swift Macros in Visual Studio Code - Google Summer Of Code 2024](https://summerofcode.withgoogle.com/programs/2024/projects/zQNf7ztP)
@lokesh-tr @ahoppen @adam-fowler

### CODE REVIEW

File: src/sourcekit-lsp/peekDocuments.ts [hunk 1]
```diff
@@ -32,10 +32,7 @@ export function activatePeekDocuments(client: langclient.LanguageClient): vscode
 
             await vscode.commands.executeCommand(
                 "editor.action.peekLocations",
-                vscode.Uri.from({
-                    scheme: "file",
-                    path: new URL(params.uri).pathname,
-                }),
+                vscode.Uri.parse(new URL(params.uri).href, true),
```
Comment (adam-fowler) [inline src/sourcekit-lsp/peekDocuments.ts:35]: Have you tried using the `protocol2CodeConverter` that comes with the vscode LSP client? Converts from LSP types to vscode types.
```
let uri = client.protocol2CodeConverter.asUri(params.uri);
```

File: src/sourcekit-lsp/getReferenceDocument.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ export function activateGetReferenceDocument(client: langclient.LanguageClient):
         {
             provideTextDocumentContent: async (uri, token) => {
                 const params: GetReferenceDocumentParams = {
-                    uri: uri.toString(true),
+                    uri: uri.toString(),
```
Comment (adam-fowler) [inline src/sourcekit-lsp/getReferenceDocument.ts:25]: Best to use `code2protocolConverter` to convert between vscode and lsp types. See below

**Review by adam-fowler — CHANGES_REQUESTED** [review]
One minor thing about converting between vscode and lsp types

File: src/sourcekit-lsp/peekDocuments.ts (hunk 1 above)
Comment (lokesh-tr) [reply src/sourcekit-lsp/peekDocuments.ts:35]: I will look into this 😌

**Review by adam-fowler — APPROVED** [review]
Looks good

**Pull request merged**